zksend/zk_bag/sources/zk_bag.move

```diff
@@ -34,6 +35,45 @@ const EClaimAddressNotExists: u64 = 6;
 /// Claims an item that does not exist.
 const EItemNotExists: u64 = 7;
 
+
+/// Event emitted when a new ZkBag is created
+public struct BagCreatedEvent has copy, drop {
+    bag_id: ID,
+    creator: address,
+}
+
+/// Event emitted when an item of type T is added to a ZkBag
+public struct BagItemAddedEvent<phantom T> has copy, drop {
+    bag_id: ID,
+    creator: address,
+}
+
+/// Event emitted when an item of type T is claimed from a ZkBag
+public struct BagItemClaimedEvent<phantom T> has copy, drop {
+    bag_id: ID,
+    creator: address,
+    receiver: address,
+}
+
+/// Event emitted when all items in a ZkBag are claimed
+public struct BagClaimedEvent has copy, drop {
+    bag_id: ID,
+    creator: address,
+    receiver: address,
+}
+
+/// Event emitted when ownership of a ZkBag is transferred to a new address
+public struct BagOwnerUpdatedEvent has copy, drop {
+    bag_id: ID,
+    old_owner: address,
+    new_owner: address,
+}
+
+/// Event emitted when a ZkBag is destroyed after all items are claimed
+public struct BagDestroyedEvent has copy, drop {
+    bag_id: ID,
```

Comment:
The contract requires that all items are claimed, and the bag is destroyed in the same transaction, so I'm not sure this makes sense as a separate event